public-ui/src/app/data-browser/views/survey-chart/survey-chart-react.component.tsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+import { BaseReactWrapper } from 'app/data-browser/base-react/base-react.wrapper';
+import { AgeChartReactComponent } from 'app/data-browser/charts/chart-age/chart-age-react.component';
+import { BioSexChartReactComponent } from 'app/data-browser/charts/chart-biosex/chart-biosex-react.component';
+import { VersionChartReactComponent } from 'app/data-browser/charts/chart-version/chart-version-react.component';
+import { TooltipReactComponent } from 'app/data-browser/components/tooltip/tooltip-react.component';
+import { ErrorMessageReactComponent } from 'app/data-browser/views/error-message/error-message-react.component';
+import { GraphType } from 'app/utils/enum-defs';
+import { triggerEvent } from 'app/utils/google_analytics';
+import * as React from 'react';
+
+interface State {
+    graphToShow: string;
+    selectedChartAnalysis: any;
+    displayGraphErrorMessage: boolean;
+    isLoaded: boolean;
+}
+
+interface Props {
+    graphButtons: any;
+    question: any;
+    answer: any;
+    selectedResult: any;
+    surveyName: any;
+    searchTerm: any;
+    surveyCountAnalysis: any;
+    isCopeSurvey: any;
+    versionAnalysis: any;
+}
+
+export class SurveyChartReactComponent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+        graphToShow: this.props.isCopeSurvey ? GraphType.SurveyVersion : GraphType.BiologicalSex,
+        displayGraphErrorMessage: false,
+        isLoaded: false,
+        selectedChartAnalysis: null,
+    };
+  }
+
+  componentDidMount() {
+    this.selectGraphType(this.state.graphToShow, this.props.question, this.props.answer);
+  }
+
+  selectGraphType(g: any, q: any, answer: any) {
+    const {surveyName, searchTerm} = this.props;
+    if (answer.stratum4.toLowerCase().indexOf('more than one race') > -1) {
+      triggerEvent('conceptClick', 'More than one race /ethncitiy graph view',
+        'Expand to see graphs', surveyName + ' - Q'
+        + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+        ' - ' + g, searchTerm, null);
+    }
+    triggerEvent('conceptClick', 'View Graphs',
+      'Expand to see graphs', surveyName + ' - Q'
+      + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+      ' - ' + g, searchTerm, null);
+    q.graphToShow = g;
```

Comment:
Do we need this? Seems like we could just use `g` below (line 83) instead of `q.graphToShow`

Reply:
changed this thanks